### PR TITLE
Fix `pytest --pyargs napari` test on pip install.  Add CI test

### LINKED ISSUE #2108: pytest --pyargs is broken by test_examples.py
## To Reproduce

1. `pip install --pre napari[pyqt5,testing]` in a fresh environment
2. run `pytest --pyargs napari`

Error:

```pytb
 $ NAPARI_POPUP_TESTS=1 pytest --pyargs napari
============================= test session starts ==============================
platform darwin -- Python 3.8.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
PyQt5 5.15.2 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
rootdir: /Users/jni
plugins: napari-plugin-engine-0.1.9, timeout-1.4.2, qt-3.3.0
collected 1610 items / 1 error / 1609 selected

==================================== ERRORS ====================================
_ ERROR collecting conda/envs/napari-043rc/lib/python3.8/site-packages/napari/_tests/test_examples.py _
conda/envs/napari-043rc/lib/python3.8/site-packages/_pytest/python.py:1309: in _idval
    generated_id = idfn(val)
conda/envs/napari-043rc/lib/python3.8/site-packages/napari/_tests/test_examples.py:39: in <lambda>
    ???
conda/envs/napari-043rc/lib/python3.8/pathlib.py:1038: in __new__
    self = cls._from_parts(args, init=False)
conda/envs/napari-043rc/lib/python3.8/pathlib.py:679: in _from_parts
    drv, root, parts = self._parse_args(args)
conda/envs/napari-043rc/lib/python3.8/pathlib.py:663: in _parse_args
    a = os.fspath(a)
E   TypeError: expected str, bytes or os.PathLike object, not NotSetType
```

I'm not sure what the best fix is... The *quick* fix is to remove `test_examples.py` from the distribution. I don't think the examples themselves are even packaged?

We should also create an element in the test matrix to test in this way, to make sure we catch this in the future. I think we had one but I guess it was lost in the move to GitHub Actions. I'm not quite sure how to do this with tox, though. My guess is to add an entry `pyargs` [here](https://github.com/napari/napari/blob/6beb6153d10c13d9b52161ee4dae65068959c649/tox.ini#L89) that does `cd; pytest --pyargs napari`, and then add an entry to the test matrix that adds that tox setting? CC @tlambert03

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
fixes #2108  by
1) fixing the pytest error `TypeError: expected str, bytes or os.PathLike object, not NotSetType` reported by @jni when pytest tries to calculate test `ids` from an empty list.  (now, it will simply skip the test if the list of examples is empty)
2) adds a dedicated test to both the PR tests and "full tests" that does _not_ checkout the full repo, but rather directly pip installs the commit from github.

as a side conversation, I still think point #3 in https://github.com/napari/napari/issues/2108#issuecomment-760199377 could be interesting: we could include `napari_examples` as a separate package in our sdist, allowing something like `napari demo add_image` after pip install

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by tlambert03** [comment]
working.  [here](https://github.com/napari/napari/pull/2109/checks?check_run_id=1702380361#step:4:5) is the key line in the new test... and [here](https://github.com/napari/napari/pull/2109/checks?check_run_id=1702380361#step:4:417) is the skipped examples in that test

File: .github/workflows/test_comprehensive.yml [hunk 1]
```diff
@@ -165,3 +163,26 @@ jobs:
         with:
           filename: .github/TEST_FAIL_TEMPLATE.md
           update_existing: true
+
+  test_pip_install:
+    name: Test pip install
```
Comment (sofroniewn) [inline .github/workflows/test_comprehensive.yml]:
```suggestion
    name: ubuntu-18.04 3.8 pip install
```
Can the name be something more like this for consistency?

image

**Review by sofroniewn — APPROVED** [review]
This looks good to me, nice fix, and nice extra test! I added a minor comment about the name and a release note.

I like shipping examples too, but maybe we merge this PR first and then add that in a follow-on? Up to you @tlambert03

**Comment by tlambert03** [comment]
>  I added a minor comment about the name 

thanks! included.

> I like shipping examples too, but maybe we merge this PR first and then add that in a follow-on? Up to you @tlambert03

Yeah, I think that could be cool.  definitely a follow up (if @jni likes it)

**Comment by sofroniewn** [comment]
ok, this looks great - will merge now!

File: .github/workflows/test_comprehensive.yml [hunk 2]
```diff
@@ -129,9 +129,7 @@ jobs:
 
       - name: Install tox-conda
         if: runner.os == 'macOS' && matrix.python == '3.9'
-        # see https://github.com/napari/napari/pull/2061
-        # can use tox-conda after https://github.com/tox-dev/tox-conda/pull/59 merges 
-        run: pip install git+https://github.com/tlambert-forks/tox-conda.git
+        run: pip install tox-conda
```
Comment (jni) [inline .github/workflows/test_comprehensive.yml:132]: lovely. =D

**Comment by jni** [comment]
> > I like shipping examples too, but maybe we merge this PR first and then add that in a follow-on? Up to you @tlambert03
> 
> Yeah, I think that could be cool. definitely a follow up (if @jni likes it)

As long as it's in its own 'extras' package, ie `pip install napari[demos]`... It would be obnoxious otherwise and also we are very lax with requirements with our examples...

Even with that, though, I'm not sure it's the best way to distribute examples... Typically people want to look at code in examples, as well as run it, not just run it. So I feel like a web gallery is a better format. Of course, `napari demo [name]` could also print the code to the console, or copy it to the clipboard 😂. But we are getting very fancy for some rather niche value...

**Comment by tlambert03** [comment]
> As long as it's in its own 'extras' package, ie `pip install napari[demos]`

I'm not sure it can be an extra (that would be for external deps)... I was thinking more something like an additional _package_ (something to the "find_packages" parameter in setuptools).  Installed as `_napari_demos`.

> So I feel like a web gallery is a better format

100% ... but both would be possible.

> But we are getting very fancy for some rather niche value...

yep 😂 let's ditch it

**Pull request merged**